### PR TITLE
refactor: Moved the escapeX functions from discord.js to @discord.js/formatters

### PR DESCRIPTION
**Pull request by Cl00e9ment**
## Changes

- moved the escapeX functions from `discord.js` to `@discord.js/formatters`
- converted the code from JS to TS (including JSDoc and TSDoc)
- made the linter happy
- modified the `escapeHeading`'s RegExp to pass the linter's RegExp safety test
- `escapeBulletedList` now conserves the bullet style (`-` or `*`)

## Reason

`@discord.js/formatters` provides the `italic`, `bold`, etc... functions so it's natural to have the `escapeItalic`, `escapeBold`, etc... functions here too.

With this modification, it's possible to use these functions without having to require `discord.js`. That's why I did this PR: I'm currently trying to get rid of `discord.js` and directly use `@discord.js/ws` and other packages like `@discord.js/formatters`.

## Breaking change?

I don't know if this is a breaking change. The functions are no longer in `Util`, but importing them directly from `discord.js` should still works since it re-exports the `@discord.js/formatters` functions. Maybe there is some edge case to take into account.

About the documentation, again I don't know if there is something to update or not.

### CODE REVIEW

File: packages/formatters/src/escapers.ts [hunk 1]
```diff
@@ -0,0 +1,312 @@
...
+/**
+ * Escapes masked link characters in a string.
+ *
+ * @param text - Content to escape
+ */
+export function escapeMaskedLink(text: string): string {
+	return text.replaceAll(/\[.+]\(.+\)/gm, '\\$&');
+}
+
+/* eslint-enable prefer-named-capture-group */
```
Comment (almeidx) [inline packages/formatters/src/escapers.ts:312]:
```suggestion
```

**Review by almeidx — CHANGES_REQUESTED** [review]
Need to remove these
https://github.com/discordjs/discord.js/blob/429dbccc85cabd9986b2e8bf443bf384e4ddc61a/packages/discord.js/typings/index.d.ts#L3033-L3045

**Review by vladfrangu — APPROVED** [review]

**Review by Jiralite — CHANGES_REQUESTED** [review]
#8945 should be factored into this.

**Comment by Cl00e9ment** [comment]
I added the changes made by @almeidx in #8945 and rebased the branch.

**Comment by Cl00e9ment** [comment]
oops I messed up the reviewer list. I don't know if I can add again the reviewers.

**Comment by almeidx** [comment]
This needs a rebase

**Comment by Cl00e9ment** [comment]
> This needs a rebase

done

**Review by almeidx — APPROVED** [review]

**Review by iCrawl — APPROVED** [review]

**Review by kyranet — APPROVED** [review]

**Review by SpaceEEC — APPROVED** [review]

**Pull request merged**